src/kernel-install/90-loaderentry.install

```diff
@@ -108,7 +108,7 @@ shift "$INITRD_OPTIONS_SHIFT"
 # All files listed as arguments, and staged files starting with "initrd" are installed as initrds.
 for initrd in "$@" "${KERNEL_INSTALL_STAGING_AREA}"/initrd*; do
     [ -f "$initrd" ] || {
-        [ "$initrd" = "${KERNEL_INSTALL_STAGING_AREA}/initrd*" ] && continue
```

Comment:
I think the original version is right? When there's no matching file, in posix shell you get the literal string including the glob back